ts/features/payments/bizEventsTransaction/store/reducers/index.ts

```diff
@@ -123,6 +155,66 @@ const reducer = (
         ...state,
         receiptDocument: pot.none
       };
+    case getType(hidePaymentsBizEventsReceiptAction.request):
+      const { filteredTransactions, removedIndex: transactionIndex } =
+        filterTransactionsByIdAndGetIndex(
+          state.transactions,
+          action.payload.transactionId
+        );
+
+      const {
+        filteredTransactions: filteredLatestTransactions,
+        removedIndex: latestTransactionIndex
+      } = filterTransactionsByIdAndGetIndex(
+        state.latestTransactions,
+        action.payload.transactionId
+      );
+
+      return {
+        ...state,
+        cancelTransactionRecord: pot.some({
+          ...(transactionIndex > -1
+            ? getTransactionByIndex(state.transactions, transactionIndex)
+            : getTransactionByIndex(
+                state.latestTransactions,
+                latestTransactionIndex
+              )),
+          index:
+            transactionIndex > -1 ? transactionIndex : latestTransactionIndex,
+          cancelType:
+            transactionIndex > -1 ? "transactions" : "latestTransactions"
+        }),
+        transactions: pot.some(filteredTransactions),
+        latestTransactions: pot.some(filteredLatestTransactions)
+      };
+
+    case getType(hidePaymentsBizEventsReceiptAction.failure): {
+      const restoreValue = pot.getOrElse(state.cancelTransactionRecord, null);
+      if (!restoreValue) {
+        return state;
+      }
+
+      const { cancelType, index, ...restoreItem } = restoreValue;
+      const isLatestTransaction = cancelType === "latestTransactions";
+
+      return {
+        ...state,
+        transactions: !isLatestTransaction
+          ? pot.map(state.transactions, transactions => [
+              ...transactions.slice(0, index),
+              restoreItem,
+              ...transactions.slice(index)
+            ])
+          : state.transactions,
+        latestTransactions: isLatestTransaction
+          ? pot.map(state.latestTransactions, latestTransactions => [
+              ...latestTransactions.slice(0, index),
+              restoreItem,
+              ...latestTransactions.slice(index)
+            ])
+          : state.latestTransactions
+      };
```

Comment:
I see that the calculation of the attribute `transactions` and `latestTransactions` is the same and what changes is only the variable where to restore the transaction.
What do you think about adding an helper function something like this:
```typescript
const restoreTransactionAtIndex = (transactionPot, restoreItem, index: number) => 
    pot.map(transactionPot, transactions => [
      ...transactions.slice(0, index),
      restoreItem,
      ...transactions.slice(index)
    ]);
 ```
 And then this block of code become:
 
 ```typescript

    transactions: cancelType !== "latestTransactions"
      ? restoreTransactionAtIndex(state.transactions, restoreItem, index)
      : state.transactions,
    latestTransactions: cancelType === "latestTransactions"
      ? restoreTransactionAtIndex(state.latestTransactions, restoreItem, index)
      : state.latestTransactions 

 ```